src/components/mission-planning/WaypointConfigPanel.vue

```diff
@@ -0,0 +1,169 @@
+            <p class="w-[140px] text-start">Altitude:</p>
+            <div class="flex w-full pr-2 h-[35px] items-center">
+              <v-text-field
+                v-model="waypointOnMissionStore.altitude"
+                type="number"
+                density="compact"
+                variant="plain"
+                hide-details
+                class="spaced-number right-aligned-input w-[60px] text-right -mr-3"
+              ></v-text-field>
+              <p class="w-[15px] text-center">m</p>
+            </div>
+          </div>
+          <v-divider class="border-black w-full" />
+          <div class="flex w-full gap-x-4 h-[30px] justify-between items-center text-[12px] text-center">
+            <p class="w-[80px] mt-1 text-start">MAVFrame:</p>
+            <v-select
+              v-model="waypointOnMissionStore.altitudeReferenceType"
+              :items="availableFrames"
+              item-title="name"
+              item-value="value"
+              hide-details
+              attach
+              class="mb-2 spaced-number right-aligned-input"
+              density="compact"
+              theme="dark"
+              variant="plain"
+            ></v-select>
+          </div>
+        </div>
+      </template>
+    </ExpansiblePanel>
+    <v-btn
+      class="absolute bg-[#00000066] text-white bottom-2 right-2"
+      variant="plain"
+      size="small"
+      prepend-icon="mdi-delete"
+      @click="handleRemoveWaypoint(selectedWaypoint)"
+    >
+      Delete
+    </v-btn>
+  </div>
+</template>
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue'
+
+import ExpansiblePanel from '@/components/ExpansiblePanel.vue'
+import { useAppInterfaceStore } from '@/stores/appInterface'
+import { useMissionStore } from '@/stores/mission'
+import { AltitudeReferenceType, Waypoint } from '@/types/mission'
+
+const interfaceStore = useAppInterfaceStore()
+const missionStore = useMissionStore()
+
+const props = defineProps<{
+  /**
+   * Waypoint to be edited
+   */
+  selectedWaypoint: Waypoint
+}>()
+
+const emit = defineEmits<{
+  (event: 'removeWaypoint', waypoint: Waypoint): void
+}>()
+
+const selectedWaypoint = ref(props.selectedWaypoint)
+
+const waypointOnMissionStore = computed(() =>
+  missionStore.currentPlanningWaypoints.find((waypoint) => waypoint.id === selectedWaypoint.value.id)
+)
+
+const editableLat = ref(selectedWaypoint.value.coordinates[0])
+const editableLng = ref(selectedWaypoint.value.coordinates[1])
+const editableAltitudeRefType = ref<AltitudeReferenceType>(
+  waypointOnMissionStore.value?.altitudeReferenceType || AltitudeReferenceType.RELATIVE_TO_HOME
+)
+
+watch(editableAltitudeRefType, (newType) => {
+  if (waypointOnMissionStore.value) {
+    waypointOnMissionStore.value.altitudeReferenceType = newType
+  }
+})
+
+const handleRemoveWaypoint = (waypoint: Waypoint): void => {
+  emit('removeWaypoint', waypoint)
+}
+
+const availableFrames = [
```

Comment:
This should be typed using the `AltitudeReferenceType` enum so we never leave one left.